cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -1311,18 +1319,24 @@ private KafkaVersionChange getKafkaVersionChange(StatefulSet kafkaSts) {
             return withVoid(serviceOperations.reconcile(namespace, zkCluster.getHeadlessServiceName(), zkHeadlessService));
         }
 
-        Future<ReconciliationState> getReconciliationStateOfConfigMap(AbstractModel cluster, ConfigMap configMap, BiFunction<Boolean, Future<ReconcileResult<ConfigMap>>, Future<ReconciliationState>> function) {
+        Future<ReconciliationState> getReconciliationStateOfConfigMap(AbstractModel cluster, ConfigMap configMap, BiFunction<ArrayList<Boolean>, Future<ReconcileResult<ConfigMap>>, Future<ReconciliationState>> function) {
```

Comment:
I think the `BiFunction<ArrayList<Boolean>,...` is pretty hairy. It would be better to declare a `static` inner class with named fields, I think.